File: java-impl/src/main/java/org/oyster/card/rule/BusFareRule.java
package org.oyster.card.rule;

import org.oyster.card.dto.JourneyType;
import org.oyster.card.dto.Zone;

import java.math.BigDecimal;

public class BusFareRule implements FareRule{
    @Override
    public boolean shouldProcess(JourneyType type, Zone one, Zone two) {
        return JourneyType.BUS.equals(type);
    }

    @Override
    public BigDecimal getFare() {
        return BigDecimal.valueOf(1.8);
    }
}

File: kotlin-impl/src/main/kotlin/org/oyster/card/rule/BusFareRule.kt
package org.oyster.card.rule

import org.oyster.card.dto.JourneyType
import org.oyster.card.dto.Zone
import java.math.BigDecimal

class BusFareRule : FareRule{

    override fun shouldProcess(type: JourneyType, one: Zone, two: Zone): Boolean {
        return JourneyType.BUS == type

    }

    override fun getFare(): BigDecimal {
        return BigDecimal.valueOf(1.80)
    }
}